kfs-core/src/main/java/edu/arizona/kfs/tax/document/IncomeTypeHandler.java

```diff
@@ -0,0 +1,576 @@
+
+		// Load the IncomeTypes from the database
+		List<IncomeType> incomeTypes = (List<IncomeType>) getBusinessObjectService().findAll(IncomeType.class);
+
+		if (incomeTypes != null) {
+			if (LOG.isDebugEnabled()) {
+				LOG.debug("found " + incomeTypes.size() + " income types");
+			}
+
+			// create IncomeType map by 1099 box
+			for (IncomeType ic : incomeTypes) {
+				if (StringUtils.isNotBlank(ic.getIncomeTypeBox())) {
+					if (LOG.isDebugEnabled()) {
+						LOG.debug("IncomeTypeCode: " + ic.getIncomeTypeCode() + ", box: " + ic.getIncomeTypeBox());
+					}
+
+					retval.put(ic.getIncomeTypeBox(), ic);
+				}
+			}
+		}
+
+		return retval;
+	}
+
+	/**
+	 * This method populates the 1099 IncomeTypes from the document accounting lines
+	 * 
+	 * @param accountingLines
+	 * @param useTaxItems
+	 */
+	public void populateIncomeTypes(List<? extends AccountingLine> accountingLines) {
+		VendorHeader vendorHeader = incomeTypeDocumentContainer.getVendorHeader();
+
+		// if we found the vendor
+		if (vendorHeader != null) {
+			// loop over the accounting lines associated with the documents and summarize
+			// valid 1099 accounting lines by income type (box on the 1099 form)
+			for (AccountingLine acctline : accountingLines) {
+				// get the 1099 box for this account
+				String box = get1099Box(vendorHeader, acctline);
+
+				if (StringUtils.isNotBlank(box)) {
+					// See if a DocumentIncomeType associated with this 1099 box
+					// exists in out map
+					T1 dit = getDocumentIncomeType(acctline, box);
+
+					if (dit != null) {
+						// add the amount from the current accounting line to the DocumentIncomeType
+						dit.setAmount(dit.getAmount().add(acctline.getAmount()));
+					}
+				}
+			}
+		}
+	}
+
+	private String get1099Box(VendorHeader vendorHeader, AccountingLine acctline) {
+		String retval = getTaxParameterHelperService().getOverridePaymentType(vendorHeader, overridePaymentTypeCodeMap);
+		if (StringUtils.isBlank(retval)) {
+			String finObjectCode = acctline.getFinancialObjectCode();
+
+			if (StringUtils.isNotBlank(finObjectCode) && objectCodeMap.containsKey(finObjectCode)) {
+				retval = objectCodeMap.get(finObjectCode);
+			} else {
+				// Set payment type to type 7 by default
+				retval = TaxConstants.AMOUNT_CODE_7;
+			}
+		}
+
+		return retval;
+	}
+
+	/**
+	 * This method checks to see if the current document is cancelled or approved - in those cases we will
+	 * not allow the income types to pre-populate
+	 * 
+	 * @return
+	 */
+	public boolean isEditableRouteStatus() {
+		String routeStatus = incomeTypeDocumentContainer.getRouteStatus();
+
+		return (KewApiConstants.ROUTE_HEADER_INITIATED_CD.equals(routeStatus) || KewApiConstants.ROUTE_HEADER_SAVED_CD.equals(routeStatus) || KewApiConstants.ROUTE_HEADER_ENROUTE_CD.equals(routeStatus));
+	}
+
+	/**
+	 * This method loops over the current DocumentIncomeType list on this document and sets the amount to 0
+	 */
+	private void zeroOutDocumentIncomeTypes() {

```

Comment:
This is now used by populateIncomeTypes above to 0 out the income types before being repopulated.